pkg/controller/deployment/util/deployment_util.go

```diff
@@ -98,6 +98,13 @@ const (
 	// MinimumReplicasUnavailable is added in a deployment when it doesn't have the minimum required replicas
 	// available.
 	MinimumReplicasUnavailable = "MinimumReplicasUnavailable"
+
+	// MaxDeploymentNameLength is the max length of deployment name.
+	MaxDeploymentNameLength = 253
```

Comment:
Do these need to be public names?  It's best if you keep them as an implementation detail rather than part of the contract.